zip-tor-txn-relay-network.rst

```diff
@@ -0,0 +1,132 @@
+::
+
+  ZIP: ??? <TTRN>
+  Title: The Zcash Tor Transaction Relay Network
+  Owners: Nathan Wilcox <nathan@electriccoin.co>
+  Credits: 
+  Status: Proposed
+  Category: Application Standard
+  Created: 2020-08-12
+  License: MIT
+
+
+Terminology
+===========
+
+The key words "MUST", "SHOULD", and "OPTIONAL" in this document are to be interpreted
+as described in RFC 2119. [#RFC2119]_
+
+The term "network upgrade" in this document is to be interpreted as described in ZIP 200
+[#zip-0200]_.
+
+The term "Sapling" in this document is to be interpreted as described in ZIP 205
+[#zip-0205]_.
+
+The term "Sprout value pool balance" in this document is to be interpreted as described
+in ZIP 209 [#zip-0209]_.
+
+
+Abstract
+========
+
+The Zcash `Tor Transaction Relay Network` (`TTRN`) is an auxillary Zcash protocol to improve network privacy of wallets which issue transactions. TTRN `Service Providers` publish a `Service Descriptor` to a well-known Wishing Well (see [FIXME link to zip-wishing-well]), which includes a `registration fee` paid into the well (ie burned). TTRN-enabled wallets subscribe to that Wishing Well to discover Providers. When sending a shielded transaction the wallet selects a random provider, connects to it over Tor, and delivers the transaction via a TTRN-specific `Delivery Protocol`. These transactions pay `TTRN service fees` to providers.
```

Comment:
I would *love* to find a robust incentivization scheme that helps both blockchain privacy and network privacy. Bonus points if it incentivizes something like Tor that is general purpose (so may provide a positive externality to those who need privacy but don't use blockchains).

It would take a lot of refinement and evolution for this kind of protocol to make that happen, and I don't even know if it's possible.

Note that there *are* multiple teams working on network privacy designs that do loop in incentives or cryptocurrency. I like Tor because it's old and battle tested and has a good pedigree.